src/Sylius/Bundle/ProductBundle/Form/DataTransformer/VariantToCombinationTransformer.php

```diff
@@ -78,16 +74,14 @@ public function reverseTransform($value)
      */
     private function matches($value)
     {
-        foreach ($this->variable->getVariants() as $variant) {
+        foreach ($this->product->getVariants() as $variant) {
             foreach ($value as $option) {
-                if (null === $option || !$variant->hasOption($option)) {
-                    continue 2;
+                if (null !== $option && $variant->hasOption($option)) {
```

Comment:
Another problem is, that if variant has more options then is provided in `value` it would be matched anyway